src/sql/base/browser/ui/table/tableCellEditorFactory.ts

```diff
@@ -96,11 +98,21 @@ export class TableCellEditorFactory {
 
 			public loadValue(item: Slick.SlickData): void {
 				this._originalValue = self._options.valueGetter(item, this._args.column) ?? '';
-				this._input.value = this._originalValue;
+				if (inputType === 'date') {
+					this._input.inputElement.valueAsDate = new Date(this._originalValue);
+				} else {
+					this._input.value = this._originalValue;
+				}
 			}
 
 			public applyValue(item: Slick.SlickData, state: string): void {
 				const activeCell = this._args.grid.getActiveCell();
+				if (inputType === 'date') {
+					// Usually, the date picker will return the date in the local time zone and change the date to the previous day.
+					// We need to convert the date to UTC time zone to avoid this behavior so that the date will be the same as the
+					// date picked in the date picker.
+					state = new Date(state).toLocaleDateString(window.navigator.language, { timeZone: 'UTC' });
```

Comment:
So this means that it'll display to the user the string value of the date they chose in UTC time? Won't that be confusing though? They would choose one date, but then the UTC date may be +- 1.

Reply:
Whenever I was converting the date from the date picker to a js Date object, the date used to be off by 1 day. This particular line of code fixes that behavior. 

This stackoverflow thread explains this: 
All of this is due to the behavior of the underlying Date.parse() trying to follow ISO 8601. When the date string follows the yyyy-mm-dd format, it's assumed to be ISO 8601 with implicit UTC 00:00. When the string deviates from the format (e.g. mm-dd-yyyy or slash instead of hyphen), it falls back to the looser parser according to RFC 2822 which uses local time when the timezone is absent. Admittedly, this will all be quite arcane to an average person.
Source: https://stackoverflow.com/questions/9509360/datepicker-date-off-by-one-day

Anyway, this does not make much difference in sts as we only get the date in string over there. So when actually query the server, the engine will consider the timezone of the server.

Reply:
Ah, that's interesting. Thanks for the context!

Did you make sure to test with different locales to make sure that Electron is picking up the language correctly? I'd also start with seeing if changing the ADS display language changes anything - but then also make sure to test changing your OS language as well in case it picks up that.